javalib/src/main/scala/java/lang/IEEE754Helpers.scala

```diff
@@ -31,21 +33,23 @@ private[java] object IEEE754Helpers {
   private def bytesToCString(bytes: Array[scala.Byte], n: Int)(implicit
       z: Zone
   ): CString = {
-    val cStr = z.alloc((n + 1).toUInt) // z.alloc() does not clear bytes.
+    val cStr = z.alloc((n + 1).toUSize) // z.alloc() does not clear bytes.
 
-    var c = 0
-    while (c < n) {
-      !(cStr + c) = bytes(c)
-      c += 1
-    }
+    // memcpy bytes from Array to CString
+    val bytesPtr = bytes.asInstanceOf[ByteArray].at(0)
+    memcpy(cStr, bytesPtr, n.toUSize)
 
+    // add NUL-terminator to CString
     !(cStr + n) = 0.toByte
 
     cStr
   }
 
   def parseIEEE754[T](s: String, f: (CString, Ptr[CString]) => T): T = {
     Zone { implicit z =>
+      if (s.isEmpty)
+        throw new NumberFormatException(exceptionMsg(s))
+
       val bytes = s.getBytes(java.nio.charset.Charset.defaultCharset())
```

Comment:
It is outside the scope of this PR, but the `defaultCharset` should  be ".UTF_8".
I will PR once this PR has merged.

Reply:
Since we expect Chars to be part of an exact set of values (digits, dots and some specific letters/symbols), why don't we convert the chars to bytes using a match/case of each Char in a while loop?

Reply:
A good and expansive question.

The string in question is resumed to be a IEEE754 floating point string.  A reading of the spec will
show that there are _lots_ of cases to consider. IIRC correctly, the handling of some tiny values
(denorm? sub-norm? Years of therapy helped me move that pain to the back of my Life.)  is particularly
delicate.

When this code was written, it was under a "Get it right!, and quickly" mandate. I was pretty sure
that the very well exercised C library code would get it right, where they define what is right!

The RYU folks might have made some improvements in parsing an IEEE string. You
could look there to see what has changed (looking at the C code is usually better than the Java
if one is looking for efficiency improvement. Sometimes the Java code picks those up, which
makes point-fix porting easier, but it is usually far behind the C code.)

You could try hand-coding something and measuring it, but I bet on the C library,
even with the call overhead, for both speed & correctness. You might try the other
color and spin the wheel.